libgadget/run.c

```diff
@@ -209,6 +211,8 @@ void run(DomainDecomp * ddecomp)
             cooling_and_starformation(&Tree);
         }
 
+        slots_free_sph_extra_data(SphP_extra);
```

Comment:
What about calling this sph_scratch_data? 
It is some data that we do not need to preserve them; the word extra doesn't convey this message. However I do have the impression we probably use 'extra' for this meaning everywhere else anyways.